### PR TITLE
Call the cb when there are no files in a directory

### LINKED ISSUE #1: Callback is not called when directory is empty
It would make this package a lot easier to use if the cb is called when the directory is empty, but just with `tree` set to an empty object (`{}`).

In my case, if the cb is not called, it keeps the promise hanging.
```js
return new Promise((resolve, reject) => {
        readFileTree(dir, { encoding: 'utf-8' }, (err, tree) => {
		if (err) reject(err);
		else resolve(tree);
	});
});
```

### PR DESCRIPTION
**Pull request by Vehmloewff**
Fixes #1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ module.exports = function readFileTree (basedir, opts, cb) {
     var parent = {}
     fs.readdir(basedir, function (err, files) {
       if (err) return cb(err)
+      if (files.length === 0) cb(null, {})
```
Comment (goto-bus-stop) [inline index.js]:
```suggestion
      if (files.length === 0) return cb(null, {})
```

**Comment by goto-bus-stop** [comment]
looks good, thanks! left a small suggestion to return early if there are no files, so that future changes in the normal code path won't accidentally call the cb twice or something.

File: index.js (hunk 1 above)
Comment (Vehmloewff) [reply index.js]: Ah, thanks for the catch!

**Comment by goto-bus-stop** [comment]
:package: 1.1.2

**Pull request merged**